### ISSUE TITLE
how to create a pre-split region table?  how to manually split/merge region?

### ISSUE THREAD

**Issue by ivanjoy42** [pingcap/tidb#6633]
how to create a pre-split region table?

how to manually split/merge region?

**Comment by tiancaiamao** [comment]
When you create a table, TiDB would allocate a new region for it automatically.

If you want to split a region manually, there is a tool for it named `pd-ctl`, in the pd/bin directory.

https://github.com/pingcap/pd/pull/947

Get region ID from table is possible using the TiDB status API:

```
curl 'http://127.0.0.1:10080//tables/{db}/{table}/regions'
```

https://github.com/pingcap/tidb/blob/master/server/http_status.go#L61

> how to manually split/merge region?

Merge region requires the two region is adjacent, use the `pd-ctl` for it.
Split region will split one region into two, so you may call it multiple times if you want split a table into several regions.

As a recap, all of those are possible for senior user but not well documented.
At this time, "table partition" is under development, we hope that new feature would ease the user in the future.

**Comment by shenli** [comment]
@ivanjoy42 Is the answer helpful for you?

**Comment by ivanjoy42** [comment]
Thank you for your help!
But there are still [problems](https://github.com/pingcap/tidb/issues/6729).

**Comment by kolbe** [comment]
@ivanjoy42 sorry you didn't get a reply until now! It looks like your ongoing problems are being addressed in issue #6729. I'm going to close this issue now, but please re-open if you have any ongoing concerns about the original topic of manually splitting/merging regions.

**Comment by tiancaiamao** [comment]
Now we also provide SQL command for this operation:

```
SPLIT TABLE table_name [INDEX index_name] BY (value_list) [, (value_list)] ...
```

https://pingcap.com/docs/stable/reference/sql/statements/split-region/
